squape/video.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2023, Cyber Alpaca
+# All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+import os
+from contextlib import contextmanager
+
+import squishinfo
+
+import test
+
+
+def _failures_results_count() -> int:
+    """
+    This function returns the total number of failures in test results by adding up
+    all the counts for errors, fatals, xpasses and fails.
+
+    Returns:
+        int: The total number of failures in test results.
+    """
+    return (
+        test.resultCount("errors")
+        + test.resultCount("fatals")
+        + test.resultCount("xpasses")
+        + test.resultCount("fails")
+    )
+
+
+def _videos_set() -> set:
+    """
+    This function returns a set of existing video filenames in video directory
+    RESULT_DIR/TEST_SUITE_NAME/TEST_CASE_NAME/attachments.
+
+    Returns:
+        set: The set of filenames with mp4 extension
+    """
+    video_dir = os.path.join(
+        squishinfo.resultDir,
+        squishinfo.testSuiteName,
+        squishinfo.testCaseName,
+        "attachments",
+    )
+
+    videos_list = set()
```

Comment:
It is called list but in fact it is a set.
```suggestion
    videos = set()
```

Reply:
Fixed with [8bbe6a6](https://github.com/CyberAlpaca/squish-api-python-extension/pull/16/commits/8bbe6a66fbea305d581cca86137158c0f0df0f9e)